src/main.cpp

```diff
@@ -103,45 +103,75 @@ parse_certs(boost::json::object const& config)
 void
 initLogging(boost::json::object const& config)
 {
+    namespace src = boost::log::sources;
+    namespace keywords = boost::log::keywords;
+    namespace sinks = boost::log::sinks;
+    namespace trivial = boost::log::trivial;
     boost::log::add_common_attributes();
     std::string format = "[%TimeStamp%] [%ThreadID%] [%Severity%] %Message%";
-    boost::log::add_console_log(
-        std::cout, boost::log::keywords::format = format);
-    if (config.contains("log_file"))
+    if (config.contains("log_to_console") &&
+        config.at("log_to_console") == true)
     {
-        boost::log::add_file_log(
-            config.at("log_file").as_string().c_str(),
-            boost::log::keywords::format = format,
-            boost::log::keywords::open_mode = std::ios_base::app);
+        boost::log::add_console_log(std::cout, keywords::format = format);
+    }
+    if (config.contains("log_to_file") && config.at("log_to_file") == true &&
+        config.contains("log_directory"))
+    {
+        boost::filesystem::path dirPath{
```

Comment:
I would add a check here that is 

```cpp
if (!config.at("log_directory").is_string())
    throw std::runtime_error("log directory must be a string");
```

or something similar. that will be more helpful than the error boost.json throws, which is 
`bad_cast` or something.

Reply:
Our config parsing happens all over the place, which is bad. We should be parsing the config on startup and throwing errors immediately if its wrong. 

#151 fixes that, tho we have to get reviews on it before we can merge...